### PR TITLE
Create nodes with hotkeys while linking

### LINKED ISSUE #1770: As a xoder I want to create essential nodes with hotkeys while linking so that I can patch faster
Follows a forum [thread](https://forum.xod.io/t/easy-access-to-pins-and-constants/2163). The following nodes arguably required more often than others:

- Terminals
- Watches and tweaks
- Constants

We can assign a hotkey which will automatically place a node under the cursor when linking has been started.

### User story

- I start linking, the rubber line is here
- I press the T key and a terminal node of the proper type with the label stollen appears under the cursor
- Likewise, C creates a constant, W creates a t**w**eak or watch

### Acceptance criteria

- [x] When linking, `T` creates a terminal under the cursor of the same type as the link
- [x] The terminal label is copied (non-normalized)
- [x] When linking an input pin, `C` creates a constant node of the proper type (or a custom type **constructor** node)
- [x] The bound value is copied to the constant node
- [x] When linking an input pin of type boolean, byte, number, or pulse, `W` creates a tweak node of the corresponding type
- [x] When linking an input pin of type string, `W` creates a `tweak-string-16` node
- [x] The bound value is copied to the tweak node
- [x] When linking an output pin, `W` creates a `watch` node

### PR DESCRIPTION
**Pull request by brusherru**
It closes #1770 

Each commit refactors a code to reduce code duplication. So probably easier to check out the whole changelog.

### CODE REVIEW

File: packages/xod-project/src/constants.js [hunk 1]
```diff
@@ -278,3 +280,10 @@ export const MANAGED_ATTACHMENT_TEMPLATES = {
   [NOT_IMPLEMENTED_IN_XOD_PATH]: IMPL_TEMPLATE,
   [TABTEST_MARKER_PATH]: TABTEST_TEMPLATE,
 };
+
+export const NODE_KIND = {
```
Comment (evgenykochetkov) [inline packages/xod-project/src/constants.js:284]: I don't think this needs to live in `xod-project`

**Review by evgenykochetkov — CHANGES_REQUESTED** [review]
Trying to create a tweak from a generic pin results in an incorrect patch.
Trying to create a constant node from a generic pin crashes the IDE.

File: packages/xod-project/src/patchPathUtils.js [hunk 2]
```diff
@@ -261,6 +262,10 @@ const constantNodeRegExp = new RegExp(
 // :: PatchPath -> Boolean
 export const isConstantNodeType = R.test(constantNodeRegExp);
 
+// :: DataType -> PatchPath
+export const getConstantNodeType = type =>
```
Comment (evgenykochetkov) [inline packages/xod-project/src/patchPathUtils.js:266]: I'd add checks for generic types here

File: packages/xod-project/src/patchPathUtils.js [hunk 3]
```diff
@@ -327,6 +332,18 @@ const tweakPathRegExp = new RegExp(
   `^xod/debug/tweak-(${bindableTypes.join('|')})`
 );
 
+// :: DataType -> Maybe PatchPath
+export const getTweakPatchPath = R.ifElse(
```
Comment (evgenykochetkov) [inline packages/xod-project/src/patchPathUtils.js:336]: and here

File: packages/xod-project/src/patchPathUtils.js (hunk 2 above)
Comment (brusherru) [reply packages/xod-project/src/patchPathUtils.js]: 👌

File: packages/xod-project/src/patchPathUtils.js (hunk 3 above)
Comment (brusherru) [reply packages/xod-project/src/patchPathUtils.js:342]: 👌

**Review by nkrkv — APPROVED** [review]
:heart:

**Review by evgenykochetkov — APPROVED** [review]

**Pull request merged**